### ISSUE TITLE
Eloquence add-on doesn't work on the log-on and secure screens

### ISSUE THREAD

**Issue by amirsol81** [fastfinge/eloquence_64#31]
@fastfinge @titet11 Now that the add-on works flawlessly, I thought I'd mention this issue. I switched from NVDA 64-Bit portable to installed, and noticed that the Eloquence add-on doesn't work on the log-on screen. So as Windows 11 starts and we enter the log-on screen to either use a fingerprint or pin to log in, NVDA uses Windows OneCore TTS instead of Eloquence. I'm also using the Use currently saved settings during sign-in and on secure screens feature from NVDA's General Settings window, but that doesn't resolve the issue.

**Comment by amirsol81** [comment]
@fastfinge Where exactly should I check? If I go to C:\Users\amirs\AppData\Roaming\nvda, I'll see no Eloquence.ini there. Just NVDA.ini.

**Comment by amirsol81** [comment]
@fastfinge I checked both places, and the only Eloquence-related file I have in both places is ECI.ini. There's no Eloquence.ini, and ECI.ini doesn't provide a parameter for that. Or maybe I'm missing something.

**Comment by fastfinge** [comment]
Right, it's ECI.ini. Now that I'm at my computer, I can give better instructions. The addon that runs when not on the logon screen is something like:
C:\Users\your-username\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence
In that folder, ECI.ini will have lines like:
Path=C:\Users\your-username\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence\ptb.syn
It will have a path= line for every single language.

Now, when you copy the addon to run as admin, it will have been moved to somewhere like:
C:\Users\admin_your-username\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence
The ECI.ini in that folder will also have lines that start with path=.
However, because the addon was copied into that folder by NVDA, instead of installed from scratch, it will not have updated them. They will say things like:
Path=C:\Users\your-username\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence\ptb.syn
When they should say things like:
Path=C:\Users\admin_your-username\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence\ptb.syn

Update the .ini file for all of the path=statements so the path is correct for every voice language. That should work.

Note that this is unsupported; you really shouldn't be running random addons from the internet on your secure logon screen. I have no idea what security issues ECI.dll might have, as it's closed source, and nobody can update or fix any that might exist.

**Comment by amirsol81** [comment]
Yes, Windows 11 Pro 25H2. And am I correct about Program Files hosting the NVDA Admin-related folder?

**Comment by amirsol81** [comment]
@fastfinge Interestingly, mine also works, sort of, on the secure screen. I mean when the secure screen appears, Eloquence still works. But as soon as I try to type my pin to log in, or as soon as I try using the fingerprint reader to log in, Eloquence disappears and another TTS engine takes over.
But if I turn on my laptop or work machine, Eloquence never appears on the Start-up screen which needs either the fingerprint reader or the password/ pin.

**Comment by fastfinge** [comment]
I think I've cracked part of this. When copying an addon for use on the secure screen, NVDA does not copy .exe files for security reasons. In your addon folder in your profile, you should have a file called eloquence_host32.exe
Example: C:\Users\samue\AppData\Roaming\nvda\addons\Eloquence\synthDrivers\eloquence_host32.exe

Could you make sure this .exe file is also included where the addons on secure screens are stored? If it's not there, copy it over. Does that change anything?

If this turns out to be the problem, it's not something I can fix, and is a security requirement of NVDA itself.
________________________________

**Comment by amirsol81** [comment]
@fastfinge So should we keep this issue open given the work-around, or should we close it?

**Comment by amirsol81** [comment]
@fastfinge I want to do that, but should I have Python Install Manager and Python 3.13 to submit a pull request for the ReadMe file?

**Comment by fastfinge** [comment]
No. That only matters if you want to build the addon for release.Sent from my iPhoneOn Dec 4, 2025, at 5:35 AM, amirsol81 ***@***.***> wrote:﻿amirsol81 left a comment (fastfinge/eloquence_64#31)
@fastfinge I want to do that, but should I have Python Install Manager and Python 3.13 to submit a pull request for the ReadMe file?

**Comment by amirsol81** [comment]
Good! So I'll handle it then!

**Comment by amirsol81** [comment]
@fastfinge I finally did it! I used the website instead of the Desktop app this time around. But the pull is there.

**Comment by amirsol81** [comment]
@fastfinge Thanks for accepting the pull. So I'm closing this issue.
